sdk/core/azure-json/src/main/java/com/azure/json/JsonNumber.java

```diff
@@ -0,0 +1,147 @@
+package com.azure.json;
+
+import java.io.IOException;
+
+/**
+ * Class representing the JSON number type
+ */
+public class JsonNumber extends JsonElement {
+    /**
+     * Stores the String representation of the current state of the JsonNumber
+     * object.
+     */
+    private Number numberValue;
+
+    /**
+     * Default constructor.
+     * Default sets numberValue to "0" through the other constructor.
+     *
+     * TODO: may need to remove this due to design guidelines? May only want to
+     * have the public JsonNumber(Number value) constructor.
+     *
+     * TODO: may need to double check that 0 is correctly cast to a Number type
+     */
+    JsonNumber() { }//this(0); }
+
+
+    JsonNumber(String value) {
+        try {
+            this.numberValue = Integer.parseInt(value);
+        } catch (Exception e) {
+            try {
+                this.numberValue = Float.parseFloat(value);
+            } catch (Exception x) {
+                x.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     * Constructor used to explicitly set the number value of the JsonNumber object
+     *
+     * @param value specifies the number this JsonNumber object represents
+     *
+     * TODO: check for invalid number values or types
+     */
+    public JsonNumber(Number value) {
+        this.numberValue = value;
+    }
+
+    /**
+     * Returns the String representation of the JsonNumber object
+     *
+     * @return the numberValue field which is a String representation of the
+     * current state of this JsonNumber object.
+     */
+    @Override
+    public String toString() {
+        try {
+            return this.numberValue.toString();
+        } catch (NullPointerException e) {
+            return null;
```

Comment:
If `null` check is done in the constructor, catching this exception won't be necessary.